smart_card_connector_app/src/chrome-api-provider.js

```diff
@@ -0,0 +1,62 @@
+goog.provide('GoogleSmartCard.ChromeAPIProvider');
+
+goog.require('GoogleSmartCard.PcscLiteServerClientsManagement.ServerRequestHandler');
+
+goog.scope(function() {
+
+const GSC = GoogleSmartCard;
+
+/**
+ * @constructor
+ * @extends goog.Disposable
+ */
+GSC.ChromeAPIProvider = function(serverMessageChannel, serverReadinessTracker) {
+  this.serverRequester_ =
+      GSC.PcscLiteServerClientsManagement.ServerRequestHandler(
+          serverMessageChannel, serverReadinessTracker, 'chrome');
+
+  chrome.smartCardProviderPrivate.onEstablishContextRequested.addListener(
+      this.webapiEstablishContextListener_.bind(this));
+
+  // TODO: addOnDisposeCallback
+};
+
+goog.inherits(GSC.ChromeAPIProvider, goog.Disposable);
+
+/** @override */
+ChromeAPIProvider.prototype.disposeInternal = function() {
+  // TODO: add impl
+};
+
+
+/**
+ * @param {number} requestId
+ */
+ChromeAPIProvider.prototype.webapiEstablishContextListener_ = function(
+    requestId) {
+  const callArguments = [2, null, null];
```

Comment:
These magic values are not great for readability... One option would be to use `third_party/pcsc-lite/naclport/js_client/src/api.js`. Another, if you'd like to avoid this dependency (although I don't think currently it'd be a big problem), would be to create some file with shared constants.